packages/muelu/example/basic/Teko.cpp

```diff
@@ -0,0 +1,504 @@
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Eric C. Cyr (eccyr@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+
+*/
+
+// Teuchos includes /*@ \label{lned:being-includes} @*/
+#include "Teuchos_ConfigDefs.hpp"
+#include "Teuchos_GlobalMPISession.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Teuchos_XMLParameterListHelpers.hpp"
+#include "Teuchos_DefaultComm.hpp"
+#include "Teuchos_AbstractFactoryStd.hpp"
+#include "Teuchos_CommandLineProcessor.hpp"
+
+// Epetra includes
+#include "mpi.h"
+
+#include "Epetra_MpiComm.h"
+#include "Epetra_CrsMatrix.h"
+#include "Epetra_Vector.h"
+
+#include "Tpetra_CrsMatrix.hpp"
+#include "Tpetra_Core.hpp"
+#include "MatrixMarket_Tpetra.hpp"
+
+// EpetraExt includes
+#include "EpetraExt_CrsMatrixIn.h"
+#include "EpetraExt_VectorIn.h"
+
+// Teko-Package includes
+#include "Teko_Utilities.hpp"
+#include "Teko_InverseFactory.hpp"
+#include "Teko_InverseLibrary.hpp"
+#include "Teko_BlockedTpetraOperator.hpp"
+#include "Teko_StridedEpetraOperator.hpp"
+#include "Teko_EpetraBlockPreconditioner.hpp"
+#include "Teko_LSCPreconditionerFactory.hpp"
+#include "Teko_InvLSCStrategy.hpp"
+#include "Teko_SIMPLEPreconditionerFactory.hpp"
+#include "Teko_TpetraInverseFactoryOperator.hpp"
+
+#include "Thyra_TpetraLinearOp.hpp"
+#include "Thyra_TpetraVectorSpace.hpp"
+
+#include "Stratimikos_DefaultLinearSolverBuilder.hpp"
+#include "Stratimikos_MueLuHelpers.hpp"
+
+#include "Thyra_Ifpack2PreconditionerFactory.hpp"
```

Comment:
Might need to check if Ifpack2 is enabled.

Reply:
I just added that to the CmakeLists.txt instead.